lib/matplotlib/tests/test_axes.py

```diff
@@ -5363,6 +5363,41 @@ def test_axisbelow():
         ax.set_axisbelow(setting)
 
 
+@image_comparison(baseline_images=['titletwiny'])
+def test_titletwiny():
+    # Test that title is put above xlabel if xlabel at top
+    fig, ax = plt.subplots()
+    fig.subplots_adjust(top=0.8)
+    ax2 = ax.twiny()
+    ax.set_xlabel('Xlabel')
+    ax2.set_xlabel('Xlabel2')
+    ax.set_title('Title')
+
+
+def test_titlesetpos():
+    # Test that title stays put if we set it manually
+    fig, ax = plt.subplots()
+    fig.subplots_adjust(top=0.8)
+    ax2 = ax.twiny()
+    ax.set_xlabel('Xlabel')
+    ax2.set_xlabel('Xlabel2')
+    ax.set_title('Title')
+    pos = (0.5, 1.11)
+    ax.title.set_position(pos)
+    renderer = fig.canvas.get_renderer()
+    ax._update_title_position(renderer)
+    assert ax.title.get_position() == pos
+
+
+def test_title_xticks_top():
+    # Test that title stays put if we set it manually
+    fig, ax = plt.subplots()
+    ax.xaxis.set_ticks_position('top')
+    ax.set_title('xlabel top')
+    fig.canvas.draw()
+    assert ax.title.get_position()[1] > 1.04
```

Comment:
Just by curiosity, is `1.04` an arbitrary threshold value that was empirically found? (I would have genuinely expected something like `> 1.0`, or maybe `> 1.01` to avoid numerical precision issues :sheep: )

Reply:
1.04 was just large enough to definitely not be at 1.0. No good reason for it.

Reply:
Fair enough, I just wanted to be sure that I was not missing something.